Wikimate.php

```diff
@@ -858,3 +858,501 @@ private function findSection($section)
 		return -1;
 	}
 }
+
+
+/**
+ * Models a wiki file that can have its properties retrieved and
+ * its contents downloaded and uploaded.
+ * All properties pertain to the current revision of the file.
+ *
+ * @author  Robert McLeod & Frans P. de Vries
+ * @since   October 2016
+ */
+class WikiFile
+{
+	protected $filename  = null;
+	protected $wikimate  = null;
+	protected $exists    = false;
+	protected $invalid   = false;
+	protected $error     = null;
+	protected $edittoken = null;
+	protected $info      = null;
+
+	/*
+	 *
+	 * Magic methods
+	 *
+	 */
+
+	/**
+	 * Constructs a WikiFile object from the filename given
+	 * and adds a Wikimate object.
```

Comment:
Well, if we're going to add documentation, we might as well make it clear. My issue is just with the "adds a Wikimate object" phrasing. Would "and associate with the passed Wikimate object" work?